File: test_brand.py
import urllib.request
import urllib.error
import urllib.parse

names = [
    "hurricane fabric.png",
    "infinity rack.png",
    "StruXure.png",
    "struxure.png",
    "StruXure/logo.png",
    "struxure/logo.png"
]
base = "https://agcxxpjkqckqqefkhazp.supabase.co/storage/v1/object/public/archivos/brands%20(logos)/"

for name in names:
    url = base + urllib.parse.quote(name)
    try:
        req = urllib.request.urlopen(url)
        print(f"{url} 200")
    except urllib.error.HTTPError as e:
        # print(f"{url} {e.code}")
        pass
    except Exception as e:
        pass

File: test_brand.js
const ids = ['struxure', 'StruXure'];

async function check() {
  const base = "https://agcxxpjkqckqqefkhazp.supabase.co/storage/v1/object/public/archivos/brands%20(logos)/";
  
  for (let id of ids) {
    // try different file names
    const names = [
      `${id}.png`,
      `${id}/logo.png`,
      `${id}/1.jpeg`,
      `${id}/1.jpg`,
      `${id}/${id}.png`,
      `${id}%20logo.png`,
      `StruXure/StruXure%20logo.png`
    ];
    for (name of names) {
      let url = base + name;
      let res = await fetch(url);
      console.log(url, res.status);
    }
  }
}
check();
